ouroboros-consensus/src/Ouroboros/Consensus/HardFork/History/Summary.hs

```diff
@@ -525,50 +516,27 @@ instance SListI xs => Serialise (Summary xs) where
   --     not yet known; the summary sent to us by the server is fine as is.
   --
   --   o @n == |xs|@: The server does not yet know about the transition out of
-  --     what (we believe to be) the final era. In principle we could decide to
-  --     leave the era summaries as-is; however, since _we_ consider that era to
-  --     be the final one, we should regard it as unbounded (it does not make
-  --     sense to have a bounded final era). We must therefore modify the final
-  --     era summary. Of course this will mean that we will make some incorrect
-  --     decisions; but as long as we aren't updated, we will anyway be unable
-  --     to deal with the next era.
+  --     what (we believe to be) the final era.
   --
   --   o @n > |xs|@: the server already knows about the transition to the next
   --     era after our final era. In this case we must drop all eras that we
-  --     don't know about, and then modify again the final era to be unbounded,
-  --     just like in the case above.
+  --     don't know about.
   --
   -- Since we do not know @|ys|@, we cannot actually implement the outermost
   -- case statement. However:
   --
   -- - If @|xs| > |ys|@, by definition @n < |xs|@, and hence we will not modify
   --   the era summary: this is what we wanted.
   --
-  -- - If @|xs| == |ys|@, then at most @n == |xs|@, in which case we might
-  --   "modify" the final era to be unbounded. But in this case, the server will
-  --   consider that era to be final as well, and so it will _already_ be
-  --   unbounded: effectively this means that this means we will leave the
-  --   summary unchanged.
+  -- - If @|xs| == |ys|@, then at most @n == |xs|@.
   decode = do
```

Comment:
I think it'd be OK to push this change without introducing a version gate and explicit backwards-compatibility logic.

* This is only used by node-to-client messaging: nodes do not send `Summary`s to other nodes.
* Specifically, nodes serialize `Summary` and clients deserialize it (via the LocalStateQuery mini protocol).
* This PR only changes the behavior of `decode`.
* We're changing it by removing a post-processing step.

The consequences of a "mismatched" node and client would be:

* if an old node is communicating with an new client, then this change is inconsequential: the old node is already sending a `Summary` that ends with an eternal Alonzo and the new client would deserialize it exactly as such (which is also how the old client would deserialize it).

* if a new node was communicating with an old client, the old client would deserialize the exact same `Summary` that it would deserialize if it were communicating with an old node. The new node would send an `Alonzo` that is not eternal (just because Alonzo is no longer the last known era -- so irrespective of this PR), but the older client would `fixEndBound` so that it was. The old node would have sent an eternal `Alonzo`, which the old client would also redundantly call `fixEndBound` on.

* Note that the previous bullet is true regardless of whether the new node has reached Babbage. If it has, then it sends a longer `Summary`, but the old client would trim off the Babbage part, leaving a non-eternal Alonzo. If it hasn't, then the node would send a `Summary` ending in a non-eternal Alonzo. Either way, the old client deserializes a `Summary` that ends with Alonzo doctored by `fixEndBound` to be eternal.

In summary: only a new client communicating with a the new node would deserialize a non-eternal Alonzo. All other configurations have the client deserializing an eternal Alonzo. So we effectively get backwards-compatibility for free, here (due to the old code's logic).

I haven't yet catalogued the ways in which the known clients (eg Cardano API, wallet, etc) use the `Summary`. I imagine some usages could give results affected by this last era. But what harm could that actually do? EG maybe it would somewhat contribute to the operator being able to _create_ transactions with invalid tx validity intervals. But -- at least before the first release after `mainnet` reaches Babbage -- that's unproblematic because of PR input-output-hk/cardano-ledger#2785: nodes will continue to accept such validity ranges in Alonzo.

Reply:
Once `mainnet` reaches Babbage, no one can be running an "old node" nor an "old client", because those things wouldn't be able to handle the Babbage txs and blocks that `mainnet` is now working with. And at that point we can considering revisiting input-output-hk/cardano-ledger#2785 to only extend the `EpochInfo` for a finite set of grandfathered-in txs.

So the possibility of this "mismatched" scenario is temporary.